### PR TITLE
Persist dataSourceId across applications under new Nav change

### PR DESCRIPTION
**Pull request by CaptainDredge**
### Description
This change persists dataSourceId across ISM sub applications when new Nav Ux is enabled

### Issues Resolved
Related Issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7027

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.ts [hunk 1]
```diff
@@ -262,12 +285,20 @@ export class IndexManagementPlugin implements Plugin<IndexManagementPluginSetup,
...
+        updater$: this.appStateUpdater,
         mount: async (params: AppMountParameters) => {
           return mountWrapper(params, ROUTES.REPOSITORIES);
         },
       });
     }
 
+    dataSourceObservable.subscribe((dataSourceOption) => {
+      if (dataSourceOption) {
+        console.log("dataSourceOption", dataSourceOption);
```
Comment (SuZhou-Joe) [inline public/plugin.ts:297]: We should remove the log here.

File: public/plugin.ts
Comment (SuZhou-Joe) [inline public/plugin.ts:1]: We need to reuse the existing application, which means we should not register the indexes and snapshot management application.

Comment (SuZhou-Joe) [inline public/plugin.ts:1]: We need to reuse the existing application, which means we should not register the indexes and snapshot management application.

File: public/plugin.ts (hunk 1 above)
Comment (CaptainDredge) [reply public/plugin.ts]: Thanks for pointing it out :)

File: public/plugin.ts
Comment (CaptainDredge) [reply public/plugin.ts:1]: Screenshot 2024-07-17 at 9 39 23 PM
Removed duplicate app registration now

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**